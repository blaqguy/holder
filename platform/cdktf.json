{
  "language": "typescript",
  "app": "npx node dist/apps/platform/main.js",
  "projectId": "67ff3db5-543b-4d0d-97f8-b195012d08a6",
  "sendCrashReports": "false",
  "codeMakerOutput": "./libs/generated/.gen/",
  "terraformProviders": [],
  "terraformModules": [
    {
      "name": "AccountFactoryForTerraform",
      "source": "aws-ia/control_tower_account_factory/aws",
      "version": "1.6.4"
    },
    {
      "name": "asg",
      "source": "terraform-aws-modules/autoscaling/aws",
      "version": "6.7.1"
    },
    {
      "name": "ecs",
      "source": "terraform-aws-modules/ecs/aws",
      "version": "5.2.0"
    }
  ]
}
