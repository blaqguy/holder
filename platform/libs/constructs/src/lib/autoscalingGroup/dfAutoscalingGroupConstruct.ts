import { AutoscalingGroup } from '@cdktf/provider-aws/lib/autoscaling-group';
import { LaunchTemplate } from '@cdktf/provider-aws/lib/launch-template';
import { Construct } from 'constructs';

export interface AutoscalingGroupProps {
  name: string;
  vpcZoneIdentifier: string[];
  desiredCapacity: number;
  maxSize: number;
  minSize: number;
  amiId: string;
  instanceType: string;
}

/**
 *  Autoscaling Group construct
 */
export class DfAutoscalingGroupConstruct extends Construct {
  protected readonly asg: AutoscalingGroup;
  /**
   *
   * @param {Construct} scope - The parent stack
   * @param {AutoscalingGroupProps} props - Properties of the autoscaling group
   */
  constructor(private scope: Construct, private props: AutoscalingGroupProps) {
    super(scope, props.name);

    const launchTemplate = new LaunchTemplate(
      this.scope,
      `${props.name}-launch-template`,
      {
        name: `${props.name}`,
        imageId: props.amiId,
        instanceType: props.instanceType,
      }
    );

    this.asg = new AutoscalingGroup(this.scope, `${props.name}-asg`, {
      name: `${props.name}`,
      desiredCapacity: props.desiredCapacity,
      maxSize: props.maxSize,
      minSize: props.minSize,
      launchTemplate: {
        name: launchTemplate.name,
      },
      vpcZoneIdentifier: props.vpcZoneIdentifier,
    });
  }
}
