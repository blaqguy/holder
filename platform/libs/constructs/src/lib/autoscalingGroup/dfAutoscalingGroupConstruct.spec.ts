import { Testing } from 'cdktf';
import { DfAutoscalingGroupConstruct } from './dfAutoscalingGroupConstruct';
import 'cdktf/lib/testing/adapters/jest';

describe('Autoscaling group deployment testing', () => {
  it('Should create an autoscaling group with a given launch template', () => {
    const synthedMockStack = Testing.synthScope((mockStack) => {
      new DfAutoscalingGroupConstruct(mockStack, {
        name: 'test',
        vpcZoneIdentifier: ['test-subnet-1', 'test-subnet-2'],
        desiredCapacity: 1,
        maxSize: 1,
        minSize: 1,
        amiId: 'test-ami-id',
        instanceType: 'test-instance-type',
      });
    });
    expect(synthedMockStack).toMatchSnapshot();
  });
});
