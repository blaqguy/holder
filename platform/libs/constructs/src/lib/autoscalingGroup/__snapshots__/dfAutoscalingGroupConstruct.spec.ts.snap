// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autoscaling group deployment testing Should create an autoscaling group with a given launch template 1`] = `
"{
  \\"resource\\": {
    \\"aws_autoscaling_group\\": {
      \\"test-asg\\": {
        \\"desired_capacity\\": 1,
        \\"launch_template\\": {
          \\"name\\": \\"\${aws_launch_template.test-launch-template.name}\\"
        },
        \\"max_size\\": 1,
        \\"min_size\\": 1,
        \\"name\\": \\"test\\",
        \\"vpc_zone_identifier\\": [
          \\"test-subnet-1\\",
          \\"test-subnet-2\\"
        ]
      }
    },
    \\"aws_launch_template\\": {
      \\"test-launch-template\\": {
        \\"image_id\\": \\"test-ami-id\\",
        \\"instance_type\\": \\"test-instance-type\\",
        \\"name\\": \\"test\\"
      }
    }
  }
}"
`;
